File: Baekjoon/Baekjoon01535.py
# Baekjoon01535.py
import sys
input = sys.stdin.readline

n = int(input())
hp = list(map(int, input().split()))
happy = list(map(int, input().split()))

dp = [ 0 for _ in range(100) ]
for i in range(n):
	for j in range(99, -1, -1):
		if (dp[j] != 0 and j + hp[i] < 100):
			dp[j + hp[i]] = max(dp[j + hp[i]], dp[j] + happy[i])

	if (hp[i] < 100):
		dp[hp[i]] = max(dp[hp[i]], happy[i])

print(max(dp))

File: Baekjoon/Baekjoon01535.cpp
// Baekjoon01535.cpp
#include <iostream>

using namespace std;

int main(void) {
	ios_base::sync_with_stdio(false);
	cin.tie(0);

	int n;
	cin >> n;

	int hp[20];
	for (int i = 0; i < n; i++)
		cin >> hp[i];

	int happy[20];
	for (int i = 0; i < n; i++)
		cin >> happy[i];

	int dp[100] = { 0, };
	for (int i = 0; i < n; i++) {
		for (int j = 99; j >= 0; j--)
			if (dp[j] && j + hp[i] < 100)
				dp[j + hp[i]] = max(dp[j + hp[i]], dp[j] + happy[i]);
		dp[hp[i]] = max(dp[hp[i]], happy[i]);
	}

	int ans = 0;
	for (int i = 0; i < 100; i++)
		ans = max(ans, dp[i]);
	cout << ans << '\n';

	return 0;
}